### DISCUSSION TITLE
Is there a recommended way to install stable and preview versions of poetry side-by-side?

### DISCUSSION THREAD

**Discussion by igor-petrik-invitae** [python-poetry/poetry#5372] [Q&A] [answered]
I would like to try out the features of the new version of poetry, but I would like to have the option to fall back to the stable version for some projects, if the preview version has a bug that impacts that package. What is the best way to achieve this, using the recommended installer?

**Comment by clintonroy** [comment, marked as answer]
The pipx installation method makes this fairly nice:

```
pipx install poetry
pipx install poetry==1.2.0b1 --suffix=b1
pipx install git+https://github.com/python-poetry/poetry.git --suffix=git
```
Gives us a poetry command (for the latest release), poetryb1 for the beta release, and poetrygit for git master.

**Reply by abn** [reply]
Just to add to that, what I personally use is a suffix `@<version>` with pipx. So I end up having commands like `poetry@1.2.0b1` etc. And putting someting like this in my bashrc helps me switch in a shell. I use the `@local` suffix for my development version.

```sh
function poetry-use() {
   version=${1:-local}
   { [ "${version}" == "local" ] && alias poetry=$HOME/workspace/python-poetry/poetry/.venv/bin/poetry; } \
	   || alias poetry=poetry@${version}
   type poetry
}

function poetry-unalias() {
   unalias poetry || :
   type poetry
}
```
